### ISSUE TITLE
React Native Android Text color not working (display only white or black color)

### ISSUE THREAD

**Issue by chinhpd1802** [facebook/react-native#32127]
Hi guys,
I just started using the react native
Currently I also have the same error as on the title: **the text color does not change, only defaults to black or white**
So can I ask why this error is happening? And what should I do to fix it?
I am connecting to android device version 8.1 and using react native 0.64.2
Thank you everyone !!

**Comment by Navipro70** [comment]
@chinhpd1802 
What do you mean, native android text color doesn't change or in your javascript code?
Give example please and steps to reproduce

**Comment by chinhpd1802** [comment]
For example, I have the following line of code:  <Text style={{ color:'red' }}>This is App</Text>
I have set the text color to red but when displayed, the text is black

**Comment by Navipro70** [comment]
Ok, color props should work normally. Give me your environment (which mac device you use), and provide complex example of code (All file). Also what about other android versions, is this bug remains?

As you can see this code working on `0.64.0`

<img width="64" alt="Снимок экрана 2021-09-01 в 15 55 12" src="https://user-images.githubusercontent.com/60893275/131674801-0c3978a6-b579-46c3-b7fb-dbb51d3884d0.png">

**Comment by chinhpd1802** [comment]
@Navipro70
I'm using a physical device: SamSung J7 Prime, android 8.1,  and  plugging it in to my computer (Win 10) using a USB cable
I tried downgrading the version to 0.64.0 but it still doesn't fix the error
This is all file: 
`import React from 'react';
import 'react-native-gesture-handler';
import { Text } from 'react-native';

const App = () => {
  return (
    <Text style={{color: 'red'}}>This is App</Text>
  );
};

export default App;
`

**Comment by Navipro70** [comment]
@chinhpd1802 
Try start emulator (Pixel 4 for example) with android studio and check again. And also wrap your text with View.

**Comment by chinhpd1802** [comment]
> @chinhpd1802
> Try start emulator (Pixel 4 for example) with android studio and check again. And also wrap your text with View.

Run on a virtual machine, the error has been fixed
But why does that error occur when using the physical  device ??

**Comment by Navipro70** [comment]
@chinhpd1802 
I'm really don't know, maybe problems with connection on your device, maybe incorrect configuration setting. But as you can see issue is not fully related to react-native

Advice you to find solution in other issues

**Comment by lcspring** [comment]
@chinhpd1802 my problem is the same as yours, how did you solve it on the physical device ?

**Comment by chinhpd1802** [comment]
@lcspring
Currently, I still don't have any solution for that error 😔😔😔

**Comment by gaoxianglyx** [comment]
same error in huawei changxiang10

**Comment by gaoxianglyx** [comment]
I find the solution.
Close android system setting => Accessibility => High contrast text, may fix it.
Open this option can make text become black which is not #ffffff color!!!

**Comment by judef** [comment]
This issue is with Android Night Mode, quick fix is to apply below under android/app/src/main/res/values/styles.xml
 
> <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
>       ...
>       <item name="android:forceDarkAllowed">false</item>
>   </style>

[SO answer here](https://stackoverflow.com/questions/57175226/how-to-disable-night-mode-in-my-application-even-if-night-mode-is-enable-in-andr/64339016#64339016)

**Comment by nilportugues** [comment]
Had the same issue, I can confirm it's the Accessibility => High contrast setting in your mobile device. Just disable that and the text will be displayed in the desired color.

This is not a bug, but an actual feature and showing it black or white is the expected behavior.

Kudos to @gaoxianglyx and I'd advice closing this issue.

**Comment by izaiasmorais** [comment]
> This issue is with Android Night Mode, quick fix is to apply below under android/app/src/main/res/values/styles.xml
> 
> ```
> <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
>     <item name="android:forceDarkAllowed">false</item>
>   </style>
> ```
> 
> [SO answer here](https://stackoverflow.com/questions/57175226/how-to-disable-night-mode-in-my-application-even-if-night-mode-is-enable-in-andr/64339016#64339016)

Thank you so much!!

**Comment by AkshaySeth94** [comment]
Has anyone been able to fix this issue?

**Comment by Yembot31013** [comment]
Yes, @AkshaySeth94 , the issue is not from react-native but from the physical device, and that is because **High contrast** is turned on, so follow this https://github.com/facebook/react-native/issues/32127#issuecomment-994480854 and it should fix it if you fall under the same category.

**Comment by TQCasey** [comment]
still this error , pls help !
React Native (0.72.10) Text style color doesn't work at all ,the text color is affected by the styles.xml (eg. <item name="android:textColor">#ffffff</item>) 

as above , all Text color are set to #ffffff .. 

pls help !

**Comment by TQCasey** [comment]
![Image](https://github.com/user-attachments/assets/4162a132-ecbb-4bf9-9db8-d76b0b9de16a)

`    render() {
        return (
            <View style={{ flex: 1, justifyContent: "center", alignItems: "center",backgroundColor : 'black' }}>
                <Text style={{ fontSize: 56, fontWeight: "bold" ,color : 'red',borderWidth : 1,borderColor : 'red',textAlign : 'center',textAlignVertical : 'center'}} >Hello World</Text>
            </View>
        );
    }`
